### ISSUE TITLE
Ad show nothing

### ISSUE THREAD

**Issue by Supamit** [floatinghotpot/cordova-admob-pro#478]
HI  i use ionic with your instrcution 
this my app.js code

//======admob code start=============
 
      var admobid = {};
        // select the right Ad Id according to platform
        if( /(android)/i.test(navigator.userAgent) ) { 
            admobid = { // for Android
                banner: 'ca-app-pub-9266638043129968/1930313335',
                interstitial: 'ca-app-pub-6869992474017983/1657046752'
            };
           
          } 

  if(window.AdMob){ 
      alert('run');
      AdMob.createBanner( {
      adId:admobid.banner, 
      position:AdMob.AD_POSITION.BOTTOM_CENTER, 
      autoShow:true} );
    
}
 
//=======AdMob Code End=======

but when i run to device nothing happen to show  
pls help me, thanks.

**Comment by mjurczyk** [comment]
Temporary fix (for iOS):

1. Remove `adId` from `createBanner/createInterstitial`, set `isTesting` to *true*. 
```js
AdMob.createBanner({
  isTesting: true
});
```

2. In plugin source files in `CDVAdMobPlugin.m` set:

```c
#define TEST_BANNER_ID           @"YOUR_AD_ID"
```

3. Go to `__buildAdRequest` in `CDVAdMobPlugin.m` and comment out the following part:

```c
}
if (
  //NSString* deviceId = [self __getAdMobDeviceId];
  //request.testDevices = [NSArray arrayWithObjects:deviceId, kGADSimulatorID, nil];
  //NSLog(@"request.testDevices: %@, <Google> tips handled", deviceId);
}
if(self.mGender) {
```

This will allow your own ads to be displayed as "test" ads.

**Comment by Supamit** [comment]
now it worked on controller.js, i don't know why it don't work with app.js.

thank you for your reply.

**Comment by mjurczyk** [comment]
@Supamit did you make sure that `deviceready` event is fired beforehand?

**Comment by Supamit** [comment]
@mjurczyk  Yes i'm sure mycode in deviceready function. it show alert popup but not show ad or prepare any space for ad, even if i try to get response but no any return both success and fail.

**Comment by HikeNH** [comment]
I'm having the same issue. In my code when I generate an ad I added an on success and on order function that both alert the result. I can see the on success always triggers with an "ok" but when I set to show the add automatically or when I call the show add function it never shows an ad. I am testing on the latest android. I have tried setting the phonegap version to 3.7 instead of the default in my config.xml but it had no effect. 

A side note. Should the app be asking for permission to show ads during installation? I'm asking because it's not asking permission to show ads and I thought I remember from the last time I used an admob plugin that it did indeed ask for permission. Could be wrong though. Hope that helps find a solution.
